src/goto-harness/recursive_initialization.cpp

```diff
@@ -325,6 +352,56 @@ std::string recursive_initialization_configt::to_string() const
         << associated_array_size.second;
   }
   out << "\n  ]";
+  out << "\n  pointer_to_treat_as_cstrings = [";
+  for(const auto &pointer_to_treat_as_string_name :
+      pointers_to_treat_as_cstrings)
+  {
+    out << "\n    " << pointer_to_treat_as_string_name << std::endl;
+  }
+  out << "\n  ]";
   out << "\n}";
   return out.str();
 }
+
+recursive_initializationt::array_convertert
+recursive_initializationt::default_array_member_initialization()
+{
+  return [this](
+    const exprt &array,
+    const std::size_t length,
+    const std::size_t current_index,
+    const std::size_t depth,
+    const recursion_sett &known_tags,
+    code_blockt &body) {
+    PRECONDITION(array.type().id() == ID_array);
+    initialize(
+      index_exprt{array, from_integer(current_index, size_type())},
+      depth,
+      known_tags,
+      body);
+  };
+}
+
+recursive_initializationt::array_convertert
+recursive_initializationt::cstring_member_initialization()
+{
+  return [this](
+    const exprt &array,
+    const std::size_t length,
+    const std::size_t current_index,
+    const std::size_t depth,
+    const recursion_sett &known_tags,
+    code_blockt &body) {
+    PRECONDITION(array.type().id() == ID_array);
+    auto const member =
+      index_exprt{array, from_integer(current_index, size_type())};
+    if(current_index + 1 == length)
+    {
+      body.add(code_assignt{member, from_integer(0, char_type())});
```

Comment:
Should use `array.type().subtype()` here, or otherwise `PRECONDITION(array.type().subtype() == char_type())`. Or even both.